### DISCUSSION TITLE
Questions about mutability in useForm

### DISCUSSION THREAD

**Discussion by jintaokoong** [mantinedev/mantine#3190] [Q&A]
I was messing with useForm for a bit and I noticed this peculiar issue that I can mutate the `values` object returned by `useForm` hook directly without using `setFieldValue` or `getInputProps`. From the repro that I've included below, you can see that when the form is submitted, values will hold the entered value even though I have not mutated it using the setters provided by the library.  I'm not sure if this is really a bug so I've decided to ask in the Q&A before submitting an issue. 

[Codesandbox repro](https://codesandbox.io/s/hardcore-bhabha-m8p857?file=/src/App.tsx)

--- EDIT ---

I think it is just me being dumb, seem like it is related to JS reference problem rather than a problem of @mantine/form

**Comment by rtivital** [comment]
Form package stores values in `useState` hook, you can do the same things with it as with regular object. If you want to mutate these values, then it's your choice.
